src/passes/GenerateDynCalls.cpp

```diff
@@ -118,6 +118,16 @@ static void exportFunction(Module& wasm, Name name, bool must_export) {
 
 void GenerateDynCalls::generateDynCallThunk(HeapType funcType) {
   Signature sig = funcType.getSignature();
+
+  if (sig.results.isTuple()) {
+    // Emscripten output is assumed to be MVP, and not to have multiple return
+    // values. In particular, signatures in Emscripten all look like "abcd"
+    // where "a" is the single return value, and "bcd" are the (in this case
+    // three) parameters.
+    Fatal() << "GenerateDynCalls: Cannot operate on multiple return values:"
+            << sig.results;
```

Comment:
Can we not check check for `--enable-mutlivalue` earlier on rather then checking for its usage like this?

Reply:
Just enabling the feature shouldn't cause an error, I think. It's ok to run `wasm-opt -all` and enable all features, for example. So this PR just checks if the multiple returns part of the multivalue feature is used in practice.

Reply:
sgtm